File: TestModels/Extendable/runtimes/python/src/simple_extendable_resources/extern/native_resource.py
# Copyright Amazon.com Inc. or its affiliates. All Rights Reserved.
# SPDX-License-Identifier: Apache-2.0

from simple_extendable_resources_internaldafny_types import (
  IExtendableResource as DafnyIExtendableResource
)
import ExtendableResource
from simple_extendable_resources.smithygenerated.models import IExtendableResource
import simple_extendable_resources_internaldafny_nativeresourcefactory

class NativeResource(IExtendableResource):
  _impl: DafnyIExtendableResource

  def __init__(self, _impl):
    self._impl = _impl

  def GetExtendableResourceData(self, nativeInput):
    return self._impl.GetExtendableResourceData(nativeInput)

  def AlwaysModeledError(self, nativeInput):
    return self._impl.AlwaysModeledError(nativeInput)

  def AlwaysMultipleErrors(self, nativeInput):
    return self._impl.AlwaysMultipleErrors(nativeInput)

  def AlwaysOpaqueError(self, nativeInput):
    if nativeInput.value == None:
      raise Exception("Python Hard Coded Exception")
    return self._impl.AlwaysOpaqueError(nativeInput)

class NativeResourceFactory:

  @staticmethod
  def DafnyFactory():
    dafny_resource = ExtendableResource.ExtendableResource()
    dafny_resource.ctor__()
    native_resource = NativeResource(dafny_resource)
    return native_resource


simple_extendable_resources_internaldafny_nativeresourcefactory.default__ = NativeResourceFactory

File: TestModels/Extendable/runtimes/java/src/main/java/Simple/Extendable/Resources/NativeResource.java
// Copyright Amazon.com Inc. or its affiliates. All Rights Reserved.
// SPDX-License-Identifier: Apache-2.0
package simple.extendable.resources;

import ExtendableResource_Compile.ExtendableResource;
import java.util.Objects;
import simple.extendable.resources.IExtendableResource;
import simple.extendable.resources.ToNative;
import simple.extendable.resources.model.GetExtendableResourceDataInput;
import simple.extendable.resources.model.GetExtendableResourceDataOutput;
import simple.extendable.resources.model.GetExtendableResourceErrorsInput;
import simple.extendable.resources.model.GetExtendableResourceErrorsOutput;

public class NativeResource implements IExtendableResource {

  private final IExtendableResource _impl;

  public NativeResource(final IExtendableResource impl) {
    this._impl = impl;
  }

  @Override
  public GetExtendableResourceDataOutput GetExtendableResourceData(
    GetExtendableResourceDataInput nativeValue
  ) {
    return this._impl.GetExtendableResourceData(nativeValue);
  }

  @Override
  public GetExtendableResourceErrorsOutput AlwaysModeledError(
    GetExtendableResourceErrorsInput nativeValue
  ) {
    return this._impl.AlwaysModeledError(nativeValue);
  }

  @Override
  public GetExtendableResourceErrorsOutput AlwaysMultipleErrors(
    GetExtendableResourceErrorsInput nativeValue
  ) {
    return this._impl.AlwaysMultipleErrors(nativeValue);
  }

  @Override
  public GetExtendableResourceErrorsOutput AlwaysOpaqueError(
    GetExtendableResourceErrorsInput nativeValue
  ) {
    if (Objects.nonNull(nativeValue.value())) {
      throw new RuntimeException("Java Hard Coded Exception");
    }
    return this._impl.AlwaysOpaqueError(nativeValue);
  }

  public static NativeResource NativeFactory() {
    ExtendableResource _nw2 = new ExtendableResource();
    _nw2.__ctor();
    IExtendableResource dafnyResource = ToNative.ExtendableResource(_nw2);
    return new NativeResource(dafnyResource);
  }
}
